packages/code-studio/src/dashboard/linker/LinkerOverlayContent.tsx

```diff
@@ -1,45 +1,78 @@
-import React, { Component } from 'react';
-import PropTypes from 'prop-types';
+import React, { Component, ErrorInfo } from 'react';
 import classNames from 'classnames';
 import { ContextActions, GLOBAL_SHORTCUTS } from '@deephaven/components';
 import Log from '@deephaven/log';
 import { PanelManager } from '../panels';
-import { UIPropTypes } from '../../include/prop-types';
 import LayoutUtils from '../../layout/LayoutUtils';
+import { Link, LinkPointUntyped } from './LinkerUtils';
 import LinkerLink from './LinkerLink';
-import LinkType from './LinkType';
 import './LinkerOverlayContent.scss';
 
 const log = Log.module('LinkerOverlayContent');
 
-export class LinkerOverlayContent extends Component {
-  constructor(props) {
+// [x,y] screen coordinates used by the Linker
+export type LinkerCoordinate = [number, number];
+
+export type VisibleLink = {
+  x1: number;
+  y1: number;
+  x2: number;
+  y2: number;
+  id: string;
+  className: string;
+};
+
+export type LinkerOverlayContentProps = {
+  disabled?: boolean;
+  links: Link[];
+  messageText: string;
+  onLinkDeleted: (linkId: string) => void;
+  onAllLinksDeleted: () => void;
+  onCancel: () => void;
+  onDone: () => void;
+  panelManager: PanelManager;
+};
+
+export type LinkerOverlayContentState = {
+  mouseX?: number;
+  mouseY?: number;
+};
+
+export class LinkerOverlayContent extends Component<
+  LinkerOverlayContentProps,
+  LinkerOverlayContentState
+> {
+  static defaultProps = {
+    disabled: 'false',
+  };
+
+  constructor(props: LinkerOverlayContentProps) {
     super(props);
 
     this.handleMouseMove = this.handleMouseMove.bind(this);
     this.handleEscapePressed = this.handleEscapePressed.bind(this);
 
     this.state = {
-      mouseX: null,
-      mouseY: null,
+      mouseX: undefined,
+      mouseY: undefined,
     };
   }
 
-  componentDidMount() {
+  componentDidMount(): void {
     window.addEventListener('mousemove', this.handleMouseMove, true);
   }
 
   // eslint-disable-next-line react/sort-comp
-  componentDidCatch(error, info) {
+  componentDidCatch?(error: Error, info: ErrorInfo): void {
```

Comment:
What is the question mark for?

Reply:
I think I copied from the d.ts and it had it there. Fixed